RandBLAS/trig_skops.hh

```diff
@@ -0,0 +1,493 @@
+#include "RandBLAS/base.hh"
+#include "RandBLAS/exceptions.hh"
+#include "RandBLAS/random_gen.hh"
+#include <RandBLAS/sparse_skops.hh>
+
+#include <Random123/philox.h>
+#include <blas.hh>
+#include <lapack.hh>
+#include <omp.h>
+
+#include <iostream>
+#include <stdio.h>
+#include <stdexcept>
+#include <string>
+#include <tuple>
+#include <random>
+
+#include <math.h>
+#include <typeinfo>
+
+#define MAX(a, b) (((a) < (b)) ? (b) : (a))
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+namespace RandBLAS {
+
+    // =============================================================================
+    /// WARNING: None of the following functions or overloads thereof are part of the
+    /// public API
+    ///
+    template<SignedInteger sint_t>
+    void generateRademacherVector_r123(sint_t* buff, uint32_t key_seed, uint32_t ctr_seed, int64_t n) {
+        typedef r123::Philox2x32 RNG;
```

Comment:
Random number generators should be template parameters, and use
```
template <typename RNG = DefaultRNG, ...>
```
where ``DefaultRNG = r123::Philox4x32`` is defined in RandBLAS/base.hh.

Reply:
Added an overload of the `generateRademacherVector_r123` method with the appropriate signature